### ISSUE TITLE
Deploy the large model locally

### ISSUE THREAD

**Issue by LMJOK** [SakanaAI/AI-Scientist#92]
Suppose I want to use the open source large model, what parameters and files do I need to set. I am now using ollama local deployment qwen2.

**Comment by LMJOK** [comment]
usage: launch_scientist.py [-h] [--skip-idea-generation] [--skip-novelty-check] [--experiment EXPERIMENT]
                           [--model {claude-3-5-sonnet-20240620,gpt-4o-2024-05-13,deepseek-coder-v2-0724,llama3.1-405b,qwen2bedrock/anthropic.claude-3-sonnet-20240229-v1:0,bedrock/anthropic.claude-3-5-sonnet-20240620-v1:0,bedrock/anthropic.claude-3-haiku-20240307-v1:0,bedrock/anthropic.claude-3-opus-20240229-v1:0vertex_ai/claude-3-opus@20240229,vertex_ai/claude-3-5-sonnet@20240620,vertex_ai/claude-3-sonnet@20240229,vertex_ai/claude-3-haiku@20240307}]
                           [--writeup {latex}] [--parallel PARALLEL] [--improvement] [--gpus GPUS] [--num-ideas NUM_IDEAS]
launch_scientist.py: error: argument --model: invalid choice: 'qwen2' (choose from 'claude-3-5-sonnet-20240620', 'gpt-4o-2024-05-13', 'deepseek-coder-v2-0724', 'llama3.1-405b', 'qwen2bedrock/anthropic.claude-3-sonnet-20240229-v1:0', 'bedrock/anthropic.claude-3-5-sonnet-20240620-v1:0', 'bedrock/anthropic.claude-3-haiku-20240307-v1:0', 'bedrock/anthropic.claude-3-opus-20240229-v1:0vertex_ai/claude-3-opus@20240229', 'vertex_ai/claude-3-5-sonnet@20240620', 'vertex_ai/claude-3-sonnet@20240229', 'vertex_ai/claude-3-haiku@20240307')

**Comment by conglu1997** [comment]
If you use Ollama, you can modify the existing OpenAI API code like so: https://ollama.com/blog/openai-compatibility

**Comment by LMJOK** [comment]
> If you use Ollama, you can modify the existing OpenAI API code like so: https://ollama.com/blog/openai-compatibility若使用 Ollama，您可按如下方式修改现有的 OpenAI API 代码：https://ollama.com/blog/openai-compatibility

Then I'm going straight to launch. Change the way the API of Ollama is invoked from the API of. py?

**Comment by LMJOK** [comment]
So I'm just going to change the way the API of Ollama is invoked from the API of launch_scientist.py?

> If you use Ollama, you can modify the existing OpenAI API code like so: https://ollama.com/blog/openai-compatibility若使用 Ollama，您可按如下方式修改现有的 OpenAI API 代码：https://ollama.com/blog/openai-compatibility

So Do I need to replace the OPENAI API calls in the code with the ollama API calls? Replace the way all models are called?

**Comment by conglu1997** [comment]
You should comment out try except so that the real errors show up. 

Also, we highly do not advise you try using Llama2 models, any model weaker than the original GPT-4 will fail.

**Comment by conglu1997** [comment]
Of course, only needs an edit here: https://github.com/SakanaAI/AI-Scientist/blob/main/ai_scientist/llm.py

**Comment by slives-lab** [comment]
Is the correct implementation shown below, or should it be more specific (e.g. ollama/deepseek-r1:7b)? Also, while ollama/deepseek isn't openAI, does should the existing OpenAI API code be modified as suggested here: https://ollama.com/blog/openai-compatibility'?

AVAILABLE_LLMS = [
    # Existing models...
    
    # Add local Ollama DeepSeek R1 model
    "ollama/deepseek-r1",
]

**Comment by conglu1997** [comment]
This should be routable to the OpenAI API, so whatever model name and creation of OpenAI client that supports that model

**Comment by slives-lab** [comment]
Using ollama deepseek, python launch_scientist.py etc launches and proceeds, but I cannot resolve the error related to: 

'Error: Failed to extract JSON from LLM output'

the images are missing, and the pdf just shows template headers - not complete. Most of the code appears to complete. Any ideas on how to fix? generated_images.png isn't generated. writeup.tex is 0KB.

**Comment by slives-lab** [comment]
It would be prudent to implement the latest ollama/deepseek-R1 models into the working code, to avoid API charges from commercial vendors, and privacy concerns with commercial vendors also.

**Comment by conglu1997** [comment]
I would recommend debugging which part fails with your integration! E.g. print debug logs from each LLM call.
